iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1395,6 +1397,33 @@ internal static string PopulateMessagePropertiesFromMessage(string topicName, Me
             return msg;
         }
 
+        private static IReadOnlyDictionary<TKey, TValue> MergeDictionaries<TKey, TValue>(IDictionary<TKey, TValue>[] dictionaries)
+        {
+            // No item in the array should be null.
+            if (dictionaries == null || dictionaries.Any(item => item == null))
+            {
+                throw new ArgumentNullException(nameof(dictionaries), "Provided dictionaries should not be null");
+            }
+
+            var result = dictionaries.SelectMany(dict => dict)
+                .ToLookup(pair => pair.Key, pair => pair.Value)
+                .ToDictionary(group => group.Key, group => group.First());
+
+            return new ReadOnlyDictionary<TKey, TValue>(result);
+        }
+
+        private static DeliveryAcknowledgement ConvertDeliveryAckTypeFromString(string value)
```

Comment:
We should consider making the enum deserializable so we don't have to maintain this mapping outside of the enum.

Reply:
This class will disappear once we move to MQTTNet, so I'm not going to bother here